client/lib/i18n-utils/test/utils.js

```diff
@@ -258,50 +300,206 @@ describe( 'utils', () => {
 		} );
 	} );
 
-	describe( '#getSupportSiteLocale', () => {
-		test( 'should return `en` by default', () => {
-			expect( getSupportSiteLocale() ).toEqual( 'en' );
+	describe( '#localizeUrl', () => {
+		test( 'should not change URL for `en`', () => {
```

Comment:
Could we add some tests for unexpected/unmatched inputs?

I'm just thinking that, in the future, someone might be tempted to do something like `localizeUrl( an.awesome.url.variable )`, where for whatever reason the variable is not a string.

In that case the last few tests fail at `url.parse( fullUrl )`:

```
		test.only( 'break me', () => {
			getLocaleSlug.mockImplementationOnce( () => 'en' );
			expect( localizeUrl( 'http://wordpress.com/tos/' ) ).toEqual( 'http://wordpress.com/tos/' );
			getLocaleSlug.mockImplementationOnce( () => 'de' );
			expect( localizeUrl( '//wordpress.com/tos/' ) ).toEqual(
				'//wordpress.com/tos/'
			);
			getLocaleSlug.mockImplementationOnce( () => 'pl' );
			expect( localizeUrl( 'wordpress.com/tos/' ) ).toEqual( 'wordpress.com/tos/' );
			getLocaleSlug.mockImplementationOnce( () => 'th' );
			expect( localizeUrl( 'asdfadfsdf' ) ).toEqual( 'asdfadfsdf' );
			getLocaleSlug.mockImplementationOnce( () => 'en' );
// Nope
			expect( localizeUrl( undefined ) ).toEqual( undefined );
			getLocaleSlug.mockImplementationOnce( () => 'en' );
			expect( localizeUrl( null ) ).toEqual( null );
			getLocaleSlug.mockImplementationOnce( () => 'en' );
			expect( localizeUrl( 123 ) ).toEqual( 123 );
		} );
```

Reply:
Oh right, thank you for ensuring to make this more robust! I have added some more testcases.